fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java

```diff
@@ -73,17 +75,21 @@ public static void analyze(UpdateStmt updateStmt, ConnectContext session) {
             selectList.addItem(item);
         }
 
-        TableRelation tableRelation = new TableRelation(tableName);
+        Relation relation = new TableRelation(tableName);
+        if (updateStmt.getFromRelations() != null) {
```

Comment:
1. If the join table is a subquery, even a correlated query, do we support it?
```
UPDATE accounts SET contact_first_name = first_name,
                    contact_last_name = last_name
  FROM (select id from employees) e WHERE e.id = accounts.sales_person;
```

2. A table comes from CTE; do we support it?

Reply:
If not, I think we should clarify the SQL that we support, and that we do not support in the doc.

Reply:
subquery is supported, CTE is not(the document doesn't mention CTE), I will add CTE support in later PRs

Reply:
basically, it's the same as select statement